testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/interceptor/register/RegisterInterceptorViaMetaFileTest.java

```
@@ -0,0 +1,74 @@
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 2110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.jboss.as.test.integration.ejb.interceptor.register;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import javax.ejb.EJB;
+
+/**
+ * @author Jiri Bilek
+ * jbilek@redhat.com on 12/03/18.
+ * Test for WFLY-9844
+ *
+ * EJB invocation for Remote interface fails when Client Interceptor
+ * registered via META-INF/services/org.jboss.ejb.client.EJBClientInterceptor
+ */
+@RunWith(Arquillian.class)
+public class RegisterInterceptorViaMetaFileTest {
+   private static final String ARCHIVE_NAME = "test-register-interceptor";
+
+   @Deployment(name = "test-register-interceptor")
+   public static Archive<?> deploy() {
+      JavaArchive jar = ShrinkWrap.create(JavaArchive.class, ARCHIVE_NAME + ".jar");
+      jar.addPackage(TestRemote.class.getPackage());
+      jar.addPackage(TestSingleton.class.getPackage());
+      jar.addPackage(TestSLSB.class.getPackage());
+      jar.addAsManifestResource(
+            RegisterInterceptorViaMetaFileTest.class.getPackage(),
+            "org.jboss.ejb.client.EJBClientInterceptor",
+            "services/org.jboss.ejb.client.EJBClientInterceptor");
+      return jar;
+   }
+
+   @EJB
+   TestSingleton testSingleton;
+
+   @Test
+   public void testInvokeSLSBthoughSingleton() {
+      String echo = "this it testing string";
+      String sss = "";
+      try {
+         sss = testSingleton.test(echo);
+      } catch (Exception ee) {
+         Assert.fail("It is not possible to invoke SLSB through singleton, when ClientInterceptor is registered via META-INF/services");
```

This is a bad practice, if the test is going to fail with an exception just let it propagate, it will have much more info for someone trying to figure out why it failed than a generic log message.

---

Thank you for advice. The code was modified to fail with the message generated by the exception.